src/api.ts

```diff
@@ -81,16 +81,16 @@ export const apiFactory = (serviceId: number) => {
 		// Immediately respond to minimize time in the client-connect script
 		res.status(200).end();
 
-		metrics.inc(Metrics.OnlineDevices);
+		if (clientRefCount[uuid] === 0) {
+			// Only increment the device as online if it wasn't previously online
+			metrics.inc(Metrics.OnlineDevices);
+		}
 		metrics.inc(Metrics.TotalDevices);
+		clientRefCount[uuid]++;
```

Comment:
This needs to be initialized btw for anyone looking, I've just pushed up this version because I'm hitting a typescript crash with it for some reason :S

Reply:
The crash is related to `uuid` being defined below but used here

Reply:
https://github.com/microsoft/TypeScript/issues/50591